test/scale.time.tests.js

```diff
@@ -245,111 +220,131 @@ describe('Time scale tests', function() {
 		config.time.unit = 'week';
 		config.time.round = 'week';
-		var Constructor = Chart.scaleService.getScaleConstructor('time');
-		var scale = new Constructor({
-			ctx: mockContext,
-			options: config, // use default config for scale
-			chart: {
-				data: mockData
-			},
-			id: scaleID
-		});
 
-		// scale.buildTicks();
-		scale.update(400, 50);
+		var scale = createScale(mockData, config);
+		scale.update(800, 200);
 
 		// last date is feb 15 because we round to start of week
 		expect(scale.ticks).toEqual(['Dec 28, 2014', 'Jan 4, 2015', 'Jan 11, 2015', 'Jan 18, 2015', 'Jan 25, 2015', 'Feb 1, 2015', 'Feb 8, 2015', 'Feb 15, 2015']);
 	});
 
-	it('Should use the min and max options', function() {
-		var scaleID = 'myScale';
-
+	describe('when specifying limits', function() {
 		var mockData = {
-			labels: ['2015-01-01T20:00:00', '2015-01-02T20:00:00', '2015-01-03T20:00:00'], // days
+			labels: ['2015-01-01T20:00:00', '2015-01-02T20:00:00', '2015-01-03T20:00:00'],
 		};
 
-		var mockContext = window.createMockContext();
-		var config = Chart.helpers.clone(Chart.scaleService.getScaleDefaults('time'));
-		config.time.min = '2015-01-01T04:00:00';
-		config.time.max = '2015-01-05T06:00:00';
-		var Constructor = Chart.scaleService.getScaleConstructor('time');
-		var scale = new Constructor({
-			ctx: mockContext,
-			options: config, // use default config for scale
-			chart: {
-				data: mockData
-			},
-			id: scaleID
+		var config;
+		beforeEach(function() {
+			config = Chart.helpers.clone(Chart.scaleService.getScaleDefaults('time'));
 		});
 
-		scale.update(400, 50);
-		expect(scale.ticks).toEqual(['Jan 1, 2015', 'Jan 5, 2015']);
+		it('should use the min option', function() {
+			config.time.unit = 'day';
+			config.time.min = '2014-12-29T04:00:00';
+
+			var scale = createScale(mockData, config);
+			expect(scale.ticks[0]).toEqual('Dec 29, 2014');
+		});
+
+		it('should use the max option', function() {
+			config.time.unit = 'day';
+			config.time.max = '2015-01-05T06:00:00';
+
+			var scale = createScale(mockData, config);
+			expect(scale.ticks[scale.ticks.length - 1]).toEqual('Jan 5, 2015');
+		});
+
+		it('should not have overlapping ticks', function() {
+			var data = {
+				labels: ['2015-01-01', '2015-06-01'],
+			};
+			config.time.max = '2015-06-02';
+
+			var scale = createScale(data, config);
+			scale.update(1000, 200);
+
+			for (var i = 0; i < scale.ticks.length - 1; i++) {
+				var firstPixel = scale.getPixelForTick(i);
+				var secondPixel = scale.getPixelForTick(i + 1);
+				var spacing = secondPixel - firstPixel;
+				var firstWidth = scale.getLabelWidth(scale.ticks[i]);
+				var secondWidth = scale.getLabelWidth(scale.ticks[i + 1]);
+				expect(spacing).toBeGreaterThan((secondWidth / 2) + (firstWidth / 2));
+			}
+		});
 	});
 
-	it('Should use the isoWeekday option', function() {
-		var scaleID = 'myScale';
+	it('should not have overlapping ticks', function() {
```

Comment:
One (in `describe('when specifying limits'...` suite) covers the case when specifying a `time.max` while the other covers the general case. Since they are in separate suites jasmine will print different messages if/when they break.